src/main/webapp/assets-lti/js/controllers/controllers.js

```diff
@@ -5,50 +5,66 @@
 canvasSupportApp.controller('courseController', ['Course', 'Courses', 'Sections', 'Friend', 'SectionSet', 'Terms', 'focus', '$scope', '$rootScope', '$filter', function (Course, Courses, Sections, Friend, SectionSet, Terms, focus, $scope, $rootScope, $filter) {
   
   $scope.contextCourseId = $rootScope.ltiLaunch.custom_canvas_course_id;
-
-  var courseUrl ='manager/api/v1/courses/' + $rootScope.ltiLaunch.custom_canvas_course_id + '?include[]=sections&_=' + generateCurrentTimestamp();
-  Course.getCourse(courseUrl).then(function (resultCourse) {
-    if(!resultCourse.data.errors) {
-      $scope.loadingSections = true;
-      $scope.course = resultCourse.data;
-      $scope.course.addingSections = false;
-      $rootScope.termId = $scope.course.enrollment_term_id;
-      Sections.getSectionsForCourseId($scope.course.id).then(function (resultSections) {
-        $scope.loadingSections = false;
-        if(!resultSections.data.errors) {
-          $scope.course.sections =_.sortBy(resultSections.data, 'name');
-          if($scope.course.sections[0].sis_course_id) {
-            $scope.currentTermSISID = $scope.course.sections[0].sis_course_id.substring(0, 4);
-          }
-          else {
-           $scope.currentTermSISID = $scope.course.sections[0].sis_course_id; 
-          }
-          if($scope.currentTermSISID) {
-          /* adds to the scope a list of sections (by sis_section_id) that the current user can perform actions on */
-          var mPathwaysCoursesUrl = 'manager/mpathways/Instructors?user=self&termid=' + $scope.currentTermSISID;
-          //var mPathwaysCoursesUrl = 'manager/mpathways/Instructors?instructor=' + $rootScope.ltiLaunch.custom_canvas_user_login_id +'&termid=' + $scope.currentTermSISID;
-            Course.getMPathwaysCourses(mPathwaysCoursesUrl, $scope.currentTermSISID).then(function (resultMPathData) {  
-              if(!resultMPathData.data) {
-                if(Array.isArray(resultMPathData)) {
-                  $scope.mpath_courses = resultMPathData;
+  $scope.currentUserId = $rootScope.ltiLaunch.custom_canvas_user_login_id;
+  Friend.lookUpCanvasFriend($scope.currentUserId).then(function (resultLookUpCanvasUser) {
```

Comment:
Let me look into this. The callbacks still need to be nested as each call depends on the previous one.

And yes, the role check will need to be masked as well.